src/account/account.ts

```diff
@@ -59,3 +81,30 @@ export async function createUser(
 
   return { ...account, reference }
 }
+
+/**
+ * Creates a new user (version 2) and uploads the encrypted account to the network
+ *
+ * @param connection connection information for data uploading
+ * @param username FDP username
+ * @param password FDP password
+ * @param mnemonic mnemonic phrase
+ */
+export async function createUser(
+  connection: Connection,
+  username: string,
+  password: string,
+  mnemonic: string,
+): Promise<Reference> {
+  const { wallet, encryptedMnemonic } = await createUserAccount(password, mnemonic)
+  const topic = generateRandomBase64String(RANDOM_TOPIC_LENGTH)
+  const encryptedMnemonicAddress = removeZeroFromHex(
+    await writeFeedData(connection, topic, stringToBytes(encryptedMnemonic), wallet.privateKey),
+  )
+  const paddingLength = Math.floor(Math.random() * (PADDING_MAX - PADDING_MIN)) + PADDING_MIN
+  const randomPadding = generateRandomBase64String(paddingLength)
+  const encryptedAddress = encrypt(password, encryptedMnemonicAddress + randomPadding)
+  const topicPublicKey = removeZeroFromHex(wallet.publicKey) + password
+
+  return writeFeedData(connection, topicPublicKey, stringToBytes(encryptedAddress), wallet.privateKey)
```

Comment:
IMO, it should be a simple SoC instead of a feed as the mnemonic won't change and if the password do change, then it will result in a different topic.

Reply:
Yes, it makes sense. But the current implementation of go fairos uses feed. We can do it with the SOC, but then we'll need to file an issue for the go version to maintain compatibility.

Reply:
Looks like an incorrect implementation in go fairos for which I did compatibility.

If I understand correctly, the document describes such an idea. A wallet is created, its public key is stored in ENS. Using the private key, a feed is created with the topic “Topic.username” (it is not clear whether this is a static name or username should be replaced with the current username). In this topic, you need to write a reference chunk with an encrypted mnemonic phrase with the current password. When the password is changed, a new chunk will be created and its reference will be written to the topic. This feed is updated to prevent logging into an account with two passwords. Since, in fact, it can technically be done, but the feed will restrict entry with old passwords.

The current implementation uses hash(publicKey+password) as the feed topic, which will not work correctly when changing the password. The feed will not be updated, and it is possible to log in with two or more passwords.

If I understand correctly, I will fix it and create an issue for the go version.